core/ecschema-editing/src/Editing/Entities.ts

```diff
@@ -196,15 +196,18 @@ export class Entities extends ECClasses {
 
     const baseClassSchema = !baseClassKey.schemaKey.matches(entityKey.schemaKey) ? await this._schemaEditor.getSchema(baseClassKey.schemaKey) : entity.schema;
     if (baseClassSchema === undefined) {
-      return { errorMessage: `Schema Key ${baseClassKey.schemaKey.toString(true)} not found in context` };
+      return { itemKey: entityKey, errorMessage: `Schema Key ${baseClassKey.schemaKey.toString(true)} not found in context` };
     }
 
     const baseClassItem = await baseClassSchema.lookupItem<EntityClass>(baseClassKey);
     if (baseClassItem === undefined)
-      return { errorMessage: `Unable to locate base class ${baseClassKey.fullName} in schema ${baseClassSchema.fullName}.` };
+      return { itemKey: entityKey, errorMessage: `Unable to locate base class ${baseClassKey.fullName} in schema ${baseClassSchema.fullName}.` };
 
     if (baseClassItem.schemaItemType !== SchemaItemType.EntityClass)
-      return { errorMessage: `${baseClassItem.fullName} is not of type Entity Class.` };
+      return { itemKey: entityKey, errorMessage: `${baseClassItem.fullName} is not of type Entity Class.` };
+
+    if (entity.baseClass !== undefined && !await baseClassItem.is(await entity.baseClass))
```

Comment:
Question, so the new base class must be related to the old base class?  I guess this is for data integrity?